pkg/tablebackup/tablebackup.go

```diff
@@ -222,37 +271,36 @@ func (t *TableBackup) janitor() {
 			if !t.triggerArchiveTimeoutOnTable() {
 				continue
 			}
-
-			t.deltaFileAccessMutex.Lock()
-			if err := t.currentDeltaFp.Close(); err != nil {
-				log.Printf("could not close %s due to inactivity: %v", t.currentDeltaFilename, err)
-				t.deltaFileAccessMutex.Unlock()
-				continue
+			t.segmentBufferMutex.Lock()
+			if err := t.archiveCurrentSegmentOnTimeout(); err != nil {
+				log.Printf("could not write changes to %s due to inactivity", t.segmentFilename, err)
 			}
-			t.currentDeltaFp = nil
-			// unlock as soon as possible, but no sooner than clearing up current delta file pointer and name
-			fname := t.currentDeltaFilename
-			t.currentDeltaFilename = ""
-			t.deltaFileAccessMutex.Unlock()
-
-			t.archiveFilesQueue.Put(fname)
-			log.Printf("archiving %s due to archiver timeout", fname)
+			t.segmentBufferMutex.Unlock()
 		}
 	}
 }
 
+func (t *TableBackup) archiveCurrentSegmentOnTimeout() error {
+	if err := t.writeSegmentToFile(); err != nil {
+		return err
+	}
+	log.Printf("archived %s due to archiver timeout", t.segmentFilename)
+	t.startNewSegment(invalidLSN)
+	return nil
```

Comment:
let's leave blank lines before the final `return` statement, it will make code more readable.

Reply:
Yep, sorry, was merging all the staging changes without a detailed look into their formatting.